examples/adjoints/adjoints_ex7/SigmaPhysics.h

```diff
@@ -0,0 +1,55 @@
+#include "libmesh/fem_system.h"
+#include "libmesh/mesh.h"
+#include "libmesh/mesh_function.h"
+#include "libmesh/fem_physics.h"
+#include "libmesh/system.h"
+
+// Using namespace libMesh
+using namespace libMesh;
+
+// FEMSystem, TimeSolver and  NewtonSolver will handle most tasks,
+// but we must specify element residuals
+class SigmaPhysics : public FEMPhysics
+{
+public:
+  // Constructor
+  SigmaPhysics():FEMPhysics(){}
+
+  std::string & fe_family() { return _fe_family; }
+  unsigned int & fe_order() { return _fe_order; }
+  Real & k() { return _k; }
+  bool & analytic_jacobians() { return _analytic_jacobians; }
+
+    // System initialization
+  virtual void init_data (System & sys);
+
+protected:
+
+  // Context initialization
+  virtual void init_context (DiffContext & context);
+
+  // Element residual and jacobian calculations
+  // Time dependent parts
+  virtual bool element_time_derivative (bool request_jacobian,
+                                        DiffContext & context);
+
+  // Clone Physics copy constructor
+  virtual UniquePtr<DifferentiablePhysics> clone_physics();
+
+  // The parameters to solve for
+  Real _k;
+
+  // Variables to hold the computed QoIs
+  Number computed_QoI[1];
+
+  // The FE type to use
+  std::string _fe_family;
+  unsigned int _fe_order;
+
+  // Index for T variable
+  unsigned int T_var;
```

Comment:
You've got T_var here, uninitialized, and unused, while 0 is hardcoded in the physics kernel itself?  Is it possible that at some point I said that our example codes are to teach new users a lesson, and you didn't realize that I meant that literally, it wasn't a euphemism for punishment?  :laughing:

Reply:
Well, my cleanup for the example focused on the adjoints_ex7.C and the element_qoi/derivative files. Forgot to take care of the other ones.